loleaflet/src/layer/marker/TextInput.js

```diff
@@ -143,6 +144,20 @@ L.TextInput = L.Layer.extend({
 		}
 	},
 
+	_onCommandResult: function(e) {
+		if ((e.commandName === '.uno:Undo' || e.commandName === '.uno:Redo') && window.mode.isMobile()) {
+			//undoing something on mobile does not trigger any input method
+			//this causes problem in mobile working with suggestions
+			//i.e: type "than" and then select "thank" from suggestion
+			//now undo and then again select "thanks" from suggestions
+			//final output is "thans"
+			//this happens because undo doesn't change the textArea value
+			//and no other way to maintain the history
+			//So better to clean the textarea so no suggestions appear
+			this._map._textInput._textArea.value = this._map._textInput._preSpaceChar;
```

Comment:
Wow - that looks terrifying - why do we fiddle in the (very tricky) internals of the _textArea - when there must be a perfectly good helper method that clears the input and gets the cursor position right (which is vital) and so on (!?).

Reply:
Thank you for pointing out my mistakes, I have corrected the PR as your suggestions.